### DISCUSSION TITLE
Suggestion:  Claimblock selling to other players

### DISCUSSION THREAD

**Discussion by rjohnson98126** [GriefPrevention/GriefPrevention#297] [Ideas]
since you dont have a lot to do these days... 

**Command**
griefprevention:SellPlayerClaimblocks [playername] [#ofBlocks] [totalPrice]

**Permissions**
griefprevention.buysellplayerclaims

unique permissions can allow different combinations.  notably users|groups that are NOT allowed to buy or sell claims via the server, but ARE allowed to buy from (and/or sell to) other players.

**Behavior**
Player1 offers to sell Player2 a number of claimblocks for a total price via this command.   

_IF_ Player1 and Player2 both have permissions  _AND IF_ Player1 has total claimblocks of that amount or greater _THEN_ Player2 receives a message describing the offer from Player 1, and Player2 is also informed of how much money they have in their balance.   Player2 is requested to confirm or decline.

If Player2 does not have enough money they can simply decline the offer without explanation to Player1.   If Player2 accepts the request with insufficient funds, the rules of the economy regarding negative balance will apply.  If Player2 is unable to go negative by the offered amount but tries anyhow, Player1 will receive the standard declined message, while Player2 will be informed that they do not have enough money and that the offer has been automatically declined.   (this scheme would prevent Player1 from 'pinging' other players to discover their economy balances)
 
If Player2 accepts the offer and has sufficient funds and/or is able to go negative, then Player1 has #ofBlocks first deducted from their bonusclaimblocks then followed by any remaining balance from accruedclaimblocks as needed.   Player2 has their bonusclaimblocks increased by the total amount,    Player2 has money deducted, and Player1 has money added.

**Caveats** 
Requests should be ignored based on any mutes Player1 has against them, as if it were a chat.  Repeated sell requests in the face of multiple declines should trip the spam auto-ban using similar rules for repeat messages, and/or there could be a cooldown timer between sell requests with this command, perhaps a 60 second default setting.

**Comment by TomLewis** [comment]
Player 1 sells Claimblocks for money, Send money to player2, player2 buys claimblocks.

No need for this.

**Comment by rjohnson98126** [comment]
Frozenbeard, Im sorry you either dont understand what im saying, or dont understand the basic verification requirements behind Grief Prevention 

 Im not saying that youre selling claimblocks to your brother or BFF who you trust not to rip you off

Im saying one party is selling to a second party and you need a neutral mechanism to ensure that neither party is a bad actor who would VIOLATE the stated transaction agreement.
